### PR TITLE
Feat: ProxyOracle, refactoring

### PR DESCRIPTION
**Pull request by Oladiy**

### CODE REVIEW

File: contracts/interfaces/IChainlinkAggregator.sol [hunk 1]
```diff
@@ -3,9 +3,20 @@ pragma solidity 0.6.12;
 
 interface IChainlinkAggregator {
   function decimals() external view returns (uint8);
-  
+
   function latestAnswer() external view returns (int256);
 
+  function latestRoundData()
```
Comment (DZahar0v) [inline contracts/interfaces/IChainlinkAggregator.sol]: You should add another interface that will contain latestRoundData()

File: contracts/misc/ProxyOracle.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import {IProxyOracle} from './interfaces/IProxyOracle.sol';
+
+contract ProxyOracle is IProxyOracle {
+  IProxyOracle Oracle;
+
+  constructor(address _oracle) public {
+    Oracle = IProxyOracle(_oracle);
+  }
+
+  function latestRoundData()
```
Comment (DZahar0v) [inline contracts/misc/ProxyOracle.sol]: Do we need this function here?

File: tasks/dev/4_oracles.ts [hunk 3]
```diff
@@ -116,8 +81,7 @@ task('dev:deploy-oracles', 'Deploy oracles for dev environment')
       ],
       verify
     );
-    // await waitForTx(await addressesProvider.setPriceOracle(fallbackOracle.address));
-    await waitForTx(await addressesProvider.setPriceOracle(aaveOracle.address));
+    await waitForTx(await addressesProvider.setPriceOracle(fallbackOracle.address));
```
Comment (DZahar0v) [inline tasks/dev/4_oracles.ts]: It should be aaveOracle

File: tasks/migrations/aave.dev.ts [hunk 4]
```diff
@@ -20,10 +20,6 @@ task('aave:dev', 'Deploy development enviroment')
     console.log('Migration started\n');
 
     // NOTE: checking chainlink oracle
```
Comment (DZahar0v) [inline tasks/migrations/aave.dev.ts:21]: Do not remove price checking

File: tasks/migrations/aave.neon.ts [hunk 5]
```diff
@@ -23,162 +18,159 @@ task('aave:neon', 'Test scenarios on NEON')
...
-    var USDC = await getMintableERC20(reserves[0]);
-    var USDT = await getMintableERC20(reserves[1]);
-    var WETH = await getMintableERC20(reserves[2]);
-    console.log(USDC.address);
-    console.log(USDT.address);
-    console.log(WETH.address);
+    let DAI = await getMintableERC20(reserves[0]);
+    let AAVE = await getMintableERC20(reserves[1]);
+    let TUSD = await getMintableERC20(reserves[2]);
+    console.log(DAI.address);
```
Comment (DZahar0v) [inline tasks/migrations/aave.neon.ts]: Remove this log

File: tasks/migrations/aave.neon.ts [hunk 6]
```diff
@@ -23,162 +18,159 @@ task('aave:neon', 'Test scenarios on NEON')
...
 
-    await USDC.connect(user1).mint(DRE.ethers.utils.parseUnits('10', 6));
-    await USDT.connect(user2).mint(DRE.ethers.utils.parseUnits('10', 6));
+    await DAI.connect(user1).mint(DRE.ethers.utils.parseUnits('10', 6));
+    await AAVE.connect(user2).mint(DRE.ethers.utils.parseUnits('10', 6));
 
-    console.log('Intitial balances');
+    console.log('Initial balances');
     console.log(
       'User 1 USDC: ',
```
Comment (DZahar0v) [inline tasks/migrations/aave.neon.ts]: Update log text

**Review by DZahar0v — CHANGES_REQUESTED** [review]

File: tasks/migrations/aave.neon.ts (hunk 5 above)
Comment (Oladiy) [reply tasks/migrations/aave.neon.ts]: > Remove this log

Done

File: contracts/interfaces/IChainlinkAggregator.sol (hunk 1 above)
Comment (Oladiy) [reply contracts/interfaces/IChainlinkAggregator.sol]: > You should add another interface that will contain latestRoundData()

Done

File: contracts/misc/ProxyOracle.sol (hunk 2 above)
Comment (Oladiy) [reply contracts/misc/ProxyOracle.sol]: > Do we need this function here?

Done. Removed redundant function

File: tasks/dev/4_oracles.ts (hunk 3 above)
Comment (Oladiy) [reply tasks/dev/4_oracles.ts]: > It should be aaveOracle

Done

File: tasks/migrations/aave.dev.ts (hunk 4 above)
Comment (Oladiy) [reply tasks/migrations/aave.dev.ts:22]: > Do not remove price checking

Done

File: tasks/migrations/aave.neon.ts (hunk 6 above)
Comment (Oladiy) [reply tasks/migrations/aave.neon.ts]: > Update log text

Done

File: tasks/migrations/aave.dev.ts (hunk 4 above)
Comment (Oladiy) [reply tasks/migrations/aave.dev.ts:21]: > Do not remove price checking

Done

**Pull request merged**